tron/config/config_parse.py

```diff
@@ -275,43 +306,81 @@ def valid_action_name(value, config_context):
 )
 
 
+def valid_mesos_action(action, config_context):
+    required_keys = {'cpus', 'mem', 'docker_image', 'mesos_address'}
+    if action.get('executor') == schema.ExecutorTypes.mesos:
+        missing_keys = required_keys - set(action.keys())
+        if missing_keys:
+            raise ConfigError(
+                'Mesos executor for action {id} is missing these required keys: {keys}'.
+                format(
+                    id=action['name'],
+                    keys=missing_keys,
+                ),
+            )
+
+
 class ValidateAction(Validator):
     """Validate an action."""
     config_class = ConfigAction
 
     defaults = {
         'node': None,
         'requires': (),
+        'retries': None,
+        'expected_runtime': datetime.timedelta(hours=24),
         'executor': schema.ExecutorTypes.ssh,
-        'cluster': None,
```

Comment:
Yeah, I *think* this will be fine. Config parsing is only for current tronfig, so works as long as the old fields don't exist in tronfig.

For objects from state, we use restore_state, and that should just ignore fields that it doesn't use.